tests/full_node_test.cpp

```diff
@@ -955,12 +955,14 @@ TEST_F(FullNodeTest, sync_two_nodes1) {
   auto num_vertices1 = nodes[0]->getDagManager()->getNumVerticesInDag();
   auto num_vertices2 = nodes[1]->getDagManager()->getNumVerticesInDag();
   for (unsigned i = 0; i < SYNC_TIMEOUT; i++) {
-    if (num_vertices1.first > 3 && num_vertices2.first > 3 && num_vertices1 == num_vertices2) break;
+    if (nodes[0]->getTransactionManager()->getTransactionPoolSize() == 0 &&
+        nodes[1]->getTransactionManager()->getTransactionPoolSize() == 0 && num_vertices1 == num_vertices2)
+      break;
```

Comment:
is this related to your changes?

Reply:
I am not sure what triggered this tests to not pass but I got intermittent failures here where vertices count was 3. I don't understand why we had it to be over count of 3. I replaced this with checking that the transactions pool is empty which means that all the inserted transactions made it to DAG.